src/batched/dense/KokkosBatched_Gemm_Decl.hpp

```diff
@@ -469,22 +475,50 @@ int BatchedGemm(BatchedGemmHandleType *const handle, const ScalarType alpha,
                                             // for now, might need to revisit
         handle->teamSz = handle->vecLen = 8;
         constexpr int tile_m = 32, tile_n = 32, tile_k = 8;
-        if (c_m % 32 == 0) {  // No bounds checking
-          ret =
-              Impl::BatchedDblBufGemm<ArgTransA, ArgTransB, ArgBatchSzDim,
-                                      BatchedGemmHandleType, ScalarType,
-                                      AViewType, BViewType, CViewType,
-                                      BoundsCheck::No, tile_m, tile_n, tile_k>(
-                  handle, alpha, A, B, beta, C)
-                  .invoke();
-        } else {
-          ret =
-              Impl::BatchedDblBufGemm<ArgTransA, ArgTransB, ArgBatchSzDim,
-                                      BatchedGemmHandleType, ScalarType,
-                                      AViewType, BViewType, CViewType,
-                                      BoundsCheck::Yes, tile_m, tile_n, tile_k>(
-                  handle, alpha, A, B, beta, C)
-                  .invoke();
+#ifdef __CUDACC_RDC__
+        constexpr size_t alpha_in_fma_thresh = 24;
+#else
+        constexpr size_t alpha_in_fma_thresh = 64;
+#endif  // __CUDAACC_RDC__
+
+        if (c_m % 32 == 0) {                 // No bounds checking
+          if (c_m >= alpha_in_fma_thresh) {  // apply alpha in fma
+            ret = Impl::BatchedDblBufGemm<ArgTransA, ArgTransB, ArgBatchSzDim,
+                                          BatchedGemmHandleType, ScalarType,
+                                          AViewType, BViewType, CViewType,
+                                          BoundsCheck::No, AlphaTag::Yes,
+                                          AlphaTag::No, tile_m, tile_n, tile_k>(
```

Comment:
It is related to my comment above. It is confused when seeing ```AlphaTag::Yes``` and ```AlphaTag::No``` appearing together.